spinnaker-monitoring-daemon/tests/newrelic_service_test.py

```diff
@@ -0,0 +1,287 @@
+# Copyright 2019 New Relic Corporation. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import os
+import unittest
+from mock import patch
+from spectator_client import SpectatorClientHelper
+from newrelic_service import (
+    NewRelicServiceFactory,
+    NewRelicMetricsService,
+    NEWRELIC_KUBERNETES_METADATA_MAPPING,
+)
+from newrelic_telemetry_sdk import MetricClient
+
+
+class MockRequest:
+    def __init__(self, status_code=200):
+        self.status_code = status_code
+
+    def raise_for_status(self):
+        if self.status_code >= 400:
+            raise Exception("error " + str(self.status_code))
+
+
+class MockMetricClient:
+    def __init__(self, insert_key, host="metric-api.newrelic.com"):
+        self.insert_key = insert_key or ""
+        self.host = host
+        self.sent_single_items = []
+        self.sent_item_batches = []
+        self.send_should_fail = False
+        self.status_code = 200
+
+    def send_should_have_status_code(self, status_code=200):
+        """ used to set a status code other than 200 for the returned response """
+        self.status_code = status_code
+
+    def send(self, item):
+        """
+        sends a single item and returns a request, storing the item sent in last_item_sent.
+        the object also records all items sent through this method in sent_single_items
+        """
+        self.last_sent_item = item
+        self.sent_single_items.append(item)
+        return MockRequest(self.status_code)
+
+    def send_batch(self, items):
+        self.last_sent_items = items
+        self.sent_item_batches.append(items)
+        return MockRequest(self.status_code)
+
+
+def generate_options(insert_key="abc", host="not-metric-api.newrelic.com", tags=None):
+    options = {
+        "newrelic": {
+            "insert_key": insert_key,
+            "host": host,
+            "tags": tags
+        }
+    }
+    return options
+
+
+def get_gauge_metric():
+    return (
+        {
+            "applicationName": "spinnaker-monitoring",
```

Comment:
I know this is random test data, but to be clear the applicationName wont be "spinnaker-monitoring". It will be the service name (e.g. "clouddriver").